plugins/json/src/main/java/org/apache/struts2/json/JSONResult.java

```diff
@@ -212,12 +211,22 @@ protected Object readRootObject(ActionInvocation invocation) {
     }
 
     protected Object findRootObject(ActionInvocation invocation) {
+        ValueStack stack = invocation.getStack();
```

Comment:
The new logic seems to make sense, and the updated comment should make it clear what the behaviour will be in 2.6.

There might be some users with existing implementations depending on the current 2.5.x behaviour (always `peek()` if root is `null`), but they could implement a custom `JSONResult` with the old behaviour if they really needed it.